pkg/util/iptables/iptables.go

```diff
@@ -575,3 +576,256 @@ func IsNotFoundError(err error) bool {
 	}
 	return false
 }
+
+type RuleOption struct {
+	Arg     string
+	Negated bool
+}
+
+type Rule struct {
+	Modules []string
+	Jump    Chain
+	Comment string
+	Options map[string]RuleOption
+}
+
+// Split a line into space-separated arguments, handling simple quoting
+func splitLine(line string) ([]string, error) {
+	items := strings.Split(line, " ")
+	args := make([]string, 0)
+	for i := 0; i < len(items); i++ {
+		if items[i] == "" {
+			continue
+		}
+
+		if !strings.HasPrefix(items[i], "\"") {
+			args = append(args, items[i])
+			continue
+		}
+
+		// combine until we find a closing quote
+		var arg string
+		items[i] = items[i][1:]
+		for j := i; j < len(items); j++ {
+			if len(arg) > 0 {
+				arg += " "
+			}
+			arg += items[j]
+			if strings.HasSuffix(arg, "\"") {
+				args = append(args, arg[:len(arg)-1])
+				break
+			}
+			i++
+		}
+		if i == len(items) {
+			return nil, fmt.Errorf("invalid iptables rule (unterminated quote): %v", line)
+		}
+	}
+	return args, nil
+}
+
+func addRuleOption(rule *Rule, option string, ro RuleOption) error {
+	if rule.Options == nil {
+		rule.Options = make(map[string]RuleOption)
+	}
+	if _, ok := rule.Options[option]; ok {
+		return fmt.Errorf("invalid iptables rule (multiple %q args)", option)
+	}
+	rule.Options[option] = ro
+	return nil
+}
+
+type optionAttrs struct {
+	argRequired bool
+	canNegate   bool
+}
+
+func parseRule(line string, allowChains []Chain, allowChainPrefixes []string) (*Rule, Chain, error) {
+	optionAttrs := map[string]optionAttrs{
+		"-A":               {true, false},
+		"-j":               {true, false},
+		"-m":               {true, false},
+		"--comment":        {true, false},
+		"--name":           {true, false},
+		"--rcheck":         {false, true},
+		"--seconds":        {true, false},
+		"--reap":           {false, false},
+		"--rsource":        {false, false},
+		"--mark":           {true, false},
+		"-p":               {true, true},
+		"-d":               {true, true},
+		"-s":               {true, true},
+		"-i":               {true, true},
+		"-o":               {true, true},
+		"--dport":          {true, true},
+		"--dst-type":       {true, false},
+		"--set-xmark":      {true, false},
+		"--to-destination": {true, false},
+		"--set":            {false, false},
+		"--mask":           {true, false},
+		"--mode":           {true, false},
+		"--probability":    {true, false},
+		"--physdev-is-in":  {false, true},
+		"--src-type":       {true, true},
+	}
+
+	var rule Rule
+	var chain Chain
+
+	items, err := splitLine(line)
+	if err != nil {
+		return nil, "", err
+	}
+
+	var opt string
+	var negated bool
+	for i := 0; i < len(items); i++ {
+		if opt == "" {
+			// negation only valid at beginning of options
+			if items[i] == "!" {
+				negated = true
+				continue
```

Comment:
It would simplify the loop if you just did an i++ here like you do for options with arguments. (eg, then both `opt` and `negated` would be local to the loop body)